src/DevMenu.tsx

```diff
@@ -27,18 +27,38 @@ import { EntropyDev } from "./DevMenu/ui/EntropyDev";
 
 import { Exploit } from "./Exploits/Exploit";
 import { useRerender } from "./ui/React/hooks";
+import { AutoExpandContext, getAutoExpandData, setAutoExpandData } from "./ui/AutoExpand/AutoExpandContext";
 
 export function DevMenuRoot(): React.ReactElement {
+  const autoExpandContextValue = useRef({
+    data: getAutoExpandData(),
+    set: (key: string, expanded: boolean) => {
+      autoExpandContextValue.current.data[key] = expanded;
+    },
+  });
+
+  useEffect(() => {
```

Comment:
This before unmount/before unload stuff seems potentially fragile. Why not just save the state every time something is expanded?

Edit: That *is* what you're doing, so I don't understand this unmount stuff.